app/presenters/queries/content_item_presenter.rb

```diff
@@ -10,20 +10,21 @@ def self.present(content_item)
         present_many(content_items).first
       end
 
-      def self.present_many(content_item_scope)
-        new(content_item_scope).present
+      def self.present_many(content_item_scope, fields: nil)
+        new(content_item_scope, fields).present
       end
 
-      def initialize(content_item_scope)
+      def initialize(content_item_scope, fields = nil)
         self.content_item_scope = content_item_scope
+        self.fields = fields
       end
 
       def present
         scope = join_supporting_objects(content_item_scope)
         scope = select_fields(scope)
 
-        items = scope.as_json.map(&:symbolize_keys)
```

Comment:
This was the main source of the performance issues. `#as_json` is incredibly slow.